### PR TITLE
Run node container with network_mode:tailscale

### PR DESCRIPTION
**Pull request by arushibandi**
Dockerfile -> docker compose.
Run two containers, use the tailscale network interface.
This uses a tailnet I set up for us. The first time you run this you'll need an auth key, I can issue some and send over.

I couldn't get Postman setup correctly with client certificates so I tested by removing all auth and ensuring I was able to access `localhost:3000/node`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,9 +1,10 @@
 module github.com/eagraf/habitat-new
 
-go 1.21
+go 1.22.0
```
[the remark this answers is not in the document]
Comment (eagraf) [reply go.mod:3]: Is the plan to import from tailscale in the future?

**Comment by arushibandi** [comment]
Tested with Postman and local certificates -- original issue was from using the wrong certificates. Updated README as well.

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (arushibandi) [reply go.mod]: using tailscale as a sidecar doesn't need the upgrade; removed this

File: README.md [hunk 2]
```diff
@@ -23,7 +23,11 @@ make install
 
 Now you are able to start Habitat. To run the node, run one of the following:
 ```
-make run-dev          # Runs the habitat node in dev mode
+make run-dev-compose          # Runs the habitat node in dev mode
+```
+To get added to the Tailscale tailnet the first time you run Habitat in dev mode, get an auth key issued from Tailscale and run. You only need to do this once, unless you remove the volumes in `.habitat`.
+```
+TAILSCALE_AUTHKEY=<authkey> make run-dev-compose
```
Comment (eagraf) [inline README.md:30]: This got changed back to `run-dev`.

File: README.md [hunk 3]
```diff
@@ -23,7 +23,11 @@ make install
 
 Now you are able to start Habitat. To run the node, run one of the following:
 ```
-make run-dev          # Runs the habitat node in dev mode
+make run-dev-compose          # Runs the habitat node in dev mode
+```
+To get added to the Tailscale tailnet the first time you run Habitat in dev mode, get an auth key issued from Tailscale and run. You only need to do this once, unless you remove the volumes in `.habitat`.
```
Comment (eagraf) [inline README.md:28]: How do I get this auth key in Tailscale?

File: README.md [hunk 4]
```diff
@@ -23,7 +23,11 @@ make install
 
 Now you are able to start Habitat. To run the node, run one of the following:
 ```
-make run-dev          # Runs the habitat node in dev mode
+make run-dev-compose          # Runs the habitat node in dev mode
```
Comment (eagraf) [inline README.md:26]: `make run-dev` here too

**Review by eagraf — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (arushibandi) [reply README.md:28]: Discussed offline -- need to generate one from tailscale account or logged-in CLI. I'm setting up a 1Password to manage stuff like this.

**Pull request merged**